### PR TITLE
✨ feat: add onPageFinished function in Config

### PR DESCRIPTION
**Pull request by tajjacob**
- Add onPageFinished function in Config
- Use onPageFinished function to get url once page is loaded

### CODE REVIEW

File: lib/request_code.dart [hunk 1]
```diff
@@ -34,6 +34,9 @@ class RequestCode {
     await controller.setBackgroundColor(Colors.transparent);
     await controller.setUserAgent(_config.userAgent);
     await controller.loadRequest(launchUri);
+    await controller.setNavigationDelegate(NavigationDelegate(
```
Comment (josemiguelvarela) [inline lib/request_code.dart:37]: Hi  @tajjacob,

This is not working when onPageFinished function is null. Can you take a look at it please?

**Review by josemiguelvarela — DISMISSED** [review]

File: lib/request_code.dart (hunk 1 above)
Comment (tajjacob) [reply lib/request_code.dart:37]: sure. I'll let you know once I fix it.

Comment (tajjacob) [reply lib/request_code.dart]: @josemiguelvarela I added this lines to solve the issue. Please check if its working or not

```
    if (_config.onPageFinished != null) {
      await controller.setNavigationDelegate(
        NavigationDelegate(
          onPageFinished: _config.onPageFinished,
        ),
      );
    }
```

Comment (josemiguelvarela) [reply lib/request_code.dart]: Sorry for the late reply @tajjacob . I'll review it today, thanks!

**Comment by josemiguelvarela** [comment]
@tajjacob the PR looks good. Before merging it, can you resolve the conflicts with master please?

**Comment by tajjacob** [comment]
@josemiguelvarela thanks for your patience. I already resolved the conflicts. please review the pr

**Review by loetsphi — APPROVED** [review]

**Comment by WKnak** [comment]
Seems like using onPageFinished it breaks the navigation. The default behavior with default NavigationDelegate is replaced.

Does it make sense to you?

```
     if (_config.onPageFinished != null) {
      await controller.setNavigationDelegate(
        NavigationDelegate(
          onNavigationRequest: _onNavigationRequest, <-- include this to fix
          onPageFinished: _config.onPageFinished,
        ),
      );
    }
```

**Pull request merged**